metals/src/main/scala/scala/meta/internal/metals/Diagnostics.scala

```diff
@@ -243,13 +243,16 @@ final class Diagnostics(
     for {
       d <- syntaxError.get(path)
       // De-duplicate only the most common and basic syntax errors.
+      isSameMessage = all.asScala.exists(diag =>
+        diag.getRange() == d.getRange() && diag.getMessage() == d.getMessage()
+      )
       isDuplicate =
         d.getMessage.startsWith("identifier expected but") &&
```

Comment:
I wonder do we even need to check this condition. Maybe checking for a start is enough and we shouldn't show syntax errors if there is a compilation error at this very spot.

Reply:
> Maybe checking for a start is enough and we shouldn't show syntax errors if there is a compilation error at this very spot.

Do you mean if there is a compilation error at the same position as the syntax error? I guess maybe if it's the _exact same spot_ then no, but I think in general showing the syntax error is important even if it's overlapping or something with a compilation error. There is probably a lot of cases where the compilation error is maybe due to the syntax error. I'm not sure. I don't have super strong feelings either way.

Reply:
I think if there is a syntax error then it will also be shown by the compiler and probably at the same spot, so should be save to deduplicate.

Reply:
Sounds good, that make sense. 👍🏼